File: src/api/partner-url.ts
import { RequestFN } from "~/core/request-function";
import { type Partner, type SessionHandle, TabLocation } from "~/models";

export const partnerURL = async (session: SessionHandle, partner: Partner): Promise<string> => {
  const request = new RequestFN(session, "SaisieURLPartenaire", {
    _Signature_: { onglet: TabLocation.Presence },

    donnees: {
      SSO: partner.sso
    }
  });

  const response = await request.send();
  return response.data.RapportSaisie.urlSSO.V;
};

File: library/src/commonMain/kotlin/ink/literate/pawnote/api/partner-url.kt
package ink.literate.pawnote.api

import ink.literate.pawnote.core.RequestFN
import ink.literate.pawnote.models.Partner
import ink.literate.pawnote.models.SessionHandle
import ink.literate.pawnote.models.TabLocation

import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.*

suspend fun partnerURL (session: SessionHandle, partner: Partner): String {
    val request = RequestFN(session.information, "SaisieURLPartenaire", Json.encodeToString(
        buildJsonObject {
            putJsonObject("_Signature_") {
                put("onglet", TabLocation.Presence.code)
            }

            putJsonObject("donnees") {
                put("SSO", partner.sso)
            }
        }
    ))

    val response = request.send()
    return Json.parseToJsonElement(response.data).jsonObject["RapportSaisie"]!!.jsonObject["urlSSO"]!!.jsonObject["V"]!!.jsonPrimitive.content
}